Mathlib/Order/Incomparable.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Incomparability relation
+
+Two values in a preorder are incomparable whenever `¬ a ≤ b` and `¬ b ≤ a`.
+
+## Main declarations
+
+* `IncompRel`: The incomparability relation. `IncompRel r a b` means that `a` and `b` are related in
+  neither direction by `r`.
+-/
+
+open Function OrderDual
+
+variable {α : Type*} {a b c d : α}
+
+section Relation
+
+variable (r : α → α → Prop)
+
+/-- The incomparability relation `IncompRel r a b` means `¬ r a b` and `¬ r b a`. -/
+def IncompRel (a b : α) : Prop :=
+  ¬ r a b ∧ ¬ r b a
+
+@[simp]
+theorem antisymmRel_compl : AntisymmRel rᶜ = IncompRel r :=
+  rfl
+
+theorem antisymmRel_compl_apply : AntisymmRel rᶜ a b ↔ IncompRel r a b := by
+  simp
+
+@[simp]
+theorem incompRel_compl : IncompRel rᶜ = AntisymmRel r := by
+  simp [← antisymmRel_compl, AntisymmRel, compl]
+
+@[simp]
+theorem incompRel_compl_apply : IncompRel rᶜ a b ↔ AntisymmRel r a b := by
+  simp
+
+theorem antisymmRel_swap_apply : AntisymmRel (swap r) a b ↔ AntisymmRel r a b :=
+  and_comm
+
+theorem incompRel_swap : IncompRel (swap r) = IncompRel r :=
+  antisymmRel_swap rᶜ
+
+theorem incompRel_swap_apply : IncompRel (swap r) a b ↔ IncompRel r a b :=
+  antisymmRel_swap_apply rᶜ
+
+@[refl]
+theorem IncompRel.refl [IsIrrefl α r] (a : α) : IncompRel r a a :=
+  AntisymmRel.refl rᶜ a
+
+variable {r} in
+theorem IncompRel.rfl [IsIrrefl α r] {a : α} : IncompRel r a a := .refl ..
+
+instance [IsIrrefl α r] : IsRefl α (IncompRel r) where
+  refl := .refl r
+
+variable {r}
+
+@[symm]
+theorem IncompRel.symm : IncompRel r a b → IncompRel r b a :=
+  And.symm
+
+instance : IsSymm α (IncompRel r) where
+  symm _ _ := IncompRel.symm
+
+instance IncompRel.decidableRel [DecidableRel r] : DecidableRel (IncompRel r) :=
+  fun _ _ ↦ inferInstanceAs (Decidable (¬ _ ∧ ¬ _))
+
+theorem IncompRel.not_antisymmRel (h : IncompRel r a b) : ¬ AntisymmRel r a b :=
+  fun h' ↦ h.1 h'.1
+
+theorem AntisymmRel.not_incompRel (h : AntisymmRel r a b) : ¬ IncompRel r a b :=
+  fun h' ↦ h'.1 h.1
+
+@[simp]
+theorem not_incompRel [IsTotal α r] : ¬ IncompRel r a b := by
+  rw [IncompRel, not_and_or, not_not, not_not]
+  exact IsTotal.total a b
+
+theorem IncompRel.not_le [LE α] (h : IncompRel (· ≤ ·) a b) : ¬ a ≤ b := h.1
+theorem IncompRel.not_ge [LE α] (h : IncompRel (· ≤ ·) a b) : ¬ b ≤ a := h.2
+theorem LE.le.not_incompRel [LE α] (h : a ≤ b) : ¬ IncompRel (· ≤ ·) a b := fun h' ↦ h'.not_le h
+
+end Relation
+
+section Preorder
```

Comment:
But then why not do it correctly? You already have a `Preorder` section
```suggestion

end Relation

section LE

variable [LE α]

theorem IncompRel.not_le (h : IncompRel (· ≤ ·) a b) : ¬ a ≤ b := h.1
theorem IncompRel.not_ge (h : IncompRel (· ≤ ·) a b) : ¬ b ≤ a := h.2
theorem LE.le.not_incompRel (h : a ≤ b) : ¬ IncompRel (· ≤ ·) a b := fun h' ↦ h'.not_le h

end LE

section Preorder
```

Reply:
Alright, I've updated the `Antisymmetrization` file to match.